opensrp-chw-core/src/test/java/org/smartregister/chw/core/activity/CoreFamilyPlanningMemberProfileActivityTest.java

```diff
@@ -80,4 +81,52 @@ public void tearDown() {
         }
     }
 
+    @Test
+    public void testOnOptionsItemSelected() {
+        activity = Mockito.spy(activity);
+
+        MenuItem menuItem = Mockito.mock(MenuItem.class);
+        CoreConstants.JSON_FORM.setLocaleAndAssetManager(activity.getApplicationContext().getResources().getConfiguration().locale, activity.getApplicationContext().getAssets());
+
+        // back pressed
+        Mockito.doReturn(android.R.id.home).when(menuItem).getItemId();
+        Mockito.doNothing().when(activity).onBackPressed();
+
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).onBackPressed();
+
+        // start form
+        Mockito.doReturn(R.id.action_registration).when(menuItem).getItemId();
+        Mockito.doNothing().when(activity).startFormForEdit(Mockito.any(), Mockito.any());
+
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).startFormForEdit(Mockito.eq(R.string.registration_info), Mockito.any());
+
+        // remove member
+        Mockito.doReturn(R.id.action_remove_member).when(menuItem).getItemId();
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).removeMember();
+
+        // start family planning registration
+        Mockito.doReturn(R.id.action_fp_change).when(menuItem).getItemId();
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).startFamilyPlanningRegistrationActivity();
+
+        // start register
+        Mockito.doReturn(R.id.action_malaria_registration).when(menuItem).getItemId();
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).startMalariaRegister();
+
+        // Malaria diagnosis
+        Mockito.doReturn(R.id.action_malaria_followup_visit).when(menuItem).getItemId();
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).startMalariaFollowUpVisit();
+
+        // Malaria diagnosis
+        Mockito.doReturn(R.id.action_malaria_diagnosis).when(menuItem).getItemId();
+        activity.onOptionsItemSelected(menuItem);
+        Mockito.verify(activity).startHfMalariaFollowupForm();
+
+
+    }
```

Comment:
➕ on this. Splitting this would allow the test to be tied to a specific feature such that a failure of the test is tied to the specific feature/functionality (menu option) being tested unlike how it is now

Reply:
Noted